src/ShaderUtils.h

```diff
@@ -4,6 +4,8 @@
 #include "OpenGL.h"
 #include "Combiner.h"
 
+static std::string finalString;
```

Comment:
I agree it is a bad name. glShaderSource takes a pointer to the string as an argument. If the string is declared inside the function then it is destroyed when the function returns and the pointer becomes invalid. If there is some other way to do this I'm open to it, I'm not a C++ expert